sw/device/lib/base/hardened_status.h

```diff
@@ -28,17 +28,20 @@ extern "C" {
 /**
  * Hardened version of the `TRY` macro from `status.h`.
  *
+ * Additionally forces the error bit to 1 if `hardened_status_ok` does not
+ * pass.
+ *
  * @param expr_ An expression that evaluates to a `status_t`.
  * @return The enclosed OK value.
  */
-#define HARDENED_TRY(expr_)                                 \
-  ({                                                        \
-    status_t status_ = expr_;                               \
-    if (hardened_status_ok(status_) != kHardenedBoolTrue) { \
-      return status_;                                       \
-    }                                                       \
-    HARDENED_CHECK_EQ(status_.value, kHardenedBoolTrue);    \
-    status_.value;                                          \
+#define HARDENED_TRY(expr_)                                            \
+  ({                                                                   \
+    status_t status_ = expr_;                                          \
+    if (hardened_status_ok(status_) != kHardenedBoolTrue) {            \
+      return (status_t){.value = status_.value | (int32_t)0x80000000}; \
```

Comment:
I think we'll want to convert this to a specific error code here.  Imagine that for some reason the OK value of the status was 0x738 (instead of 0x739):  This error case is then going to return `0x8000_0738`, which contains `0x18` in the low 5 bits (the absl_status_code portion of the error).  This is not a valid error code and the rest of the error information (module and line number) are nonsense values.

I'd suggest converting this to an error value you aren't otherwise using (e.g. `kInternal` seems appropriate; I think you could also squint at the meaning of `kOutOfRange` or `kDataLoss` to fit here).

Reply:
But wouldn't that change all error codes to `kInternal`? I added this with the thought that I don't want a failure clause of `HARDENED_TRY` to pass a later `TRY` in tests. (This is from `lib/base/hardened_status.h`, so the cryptolib status codes aren't in scope -- however, I think cryptolib is currently the only thing that uses this library.)